jetty-ee9/jetty-ee9-nested/src/main/java/org/eclipse/jetty/ee9/nested/MultiPart.java

```diff
@@ -57,10 +56,33 @@ public interface Parser
     {
         void deleteParts();
 
-        Part getPart(String name) throws IOException;
+        jakarta.servlet.http.Part getPart(String name) throws IOException;
 
-        Collection<Part> getParts() throws IOException;
+        Collection<jakarta.servlet.http.Part> getParts() throws IOException;
 
         List<ComplianceViolation.Event> getNonComplianceWarnings();
     }
+
+    /**
+     * Expose File of MultiPart section.
+     *
+     * @deprecated this interface is not available in ee10 or newer.
+     */
+    @Deprecated(since = "12.0.12", forRemoval = true)
+    public interface Part extends jakarta.servlet.http.Part
+    {
+        /**
+         * Get the file
+         *
+         * @return the file, if any, the data has been written to.
+         */
+        File getFile();
+
+        /**
+         * Get the filename from the content-disposition.
+         *
+         * @return null or the filename
+         */
+        String getContentDispositionFilename();
```

Comment:
Isn't this already available somehow?

Reply:
I think it is `javax.servlet.http.Part.getSubmittedFileName()`

Reply:
This method doesn't need to be added, `getSubmittedFileName()` should return the same information.

Reply:
It wasn't added, it is already there, it is now just being exposed.

Reply:
yeah, I'm saying we don't need to expose it because it returns exactly the same info as the standard api method `getSubmittedFileName()`.